rktl_sim/nodes/simulator

```diff
@@ -358,5 +449,244 @@ class Simulator(object):
+        if 'randomize_pose' in car_config:
+            init_pose = None
+
+        self.car_ids[car_name] = self.sim.create_car(
+            'car', init_pose=init_pose, noise=self.car_noise, car_props=self.car_properties)
+
+        car_id = self.car_ids[car_name]
+        self.car_effort_subs[car_name] = rospy.Subscriber(
+            f'/cars/{car_name}/effort', ControlEffort,
+            self.effort_cb, callback_args=car_id)
+        self.car_cmd_subs[car_name] = rospy.Subscriber(
+            f'/cars/{car_name}/command', ControlCommand,
+            self.cmd_cb, callback_args=car_id)
+
+        if self.mode == SimulatorMode.REALISTIC:
+            self.car_pose_pubs[car_name] = rospy.Publisher(
+                f'/cars/{car_name}/pose_sync_early',
+                PoseWithCovarianceStamped, queue_size=1)
+            self.car_odom_pubs[car_name] = rospy.Publisher(
+                f'/cars/{car_name}/odom_truth', Odometry, queue_size=1)
+        elif self.mode == SimulatorMode.IDEAL:
+            self.car_odom_pubs[car_name] = rospy.Publisher(
+                f'/cars/{car_name}/odom', Odometry, queue_size=1)
+
+    return CreateCarResponse(True)
+
+
+def set_sensor_noise(self):
+    """
+    adds the sensor noise for the car and for the ball
+    @return: the sensor noises for the ball and for the car
+    """
+    sensor_noise = rospy.get_sim_param('~sensor_noise', None)
+    ball_noise = None
+    car_noise = None
+    if sensor_noise is not None and self.mode == SimulatorMode.REALISTIC:
+        ball_noise = sensor_noise.get('ball', None)
+        car_noise = sensor_noise.get('car', None)
+    self.car_noise = car_noise
+    self.ball_noise = ball_noise
+    return car_noise, ball_noise
+
+
+def generate_ball(self):
+    """
+    check if we want to randomize the ball properties (ball pos and ball vel)
+    if we want to randomize, then we will do so in the sim.py
+    @return: two paramters; the ball odometer publisher and the ball pose publisher
+    depends on the type of simulator mode we are using
+    """
+    # TODO: need to potentially chagne the init pose here
+    ball_init_pose = rospy.get_sim_param('~ball/init_pose', None)
+    ball_init_speed = rospy.get_sim_param('~ball/init_speed', None)
+
+    self.sim.create_ball('ball', init_pose=ball_init_pose,
+                         init_speed=ball_init_speed, noise=self.ball_noise)
+    if self.mode == SimulatorMode.REALISTIC:
+        ball_pose_pub = rospy.Publisher('/ball/pose_sync_early',
+                                        PoseWithCovarianceStamped, queue_size=1)
+        ball_odom_pub = rospy.Publisher('/ball/odom_truth',
+                                        Odometry, queue_size=1)
+        return ball_pose_pub, ball_odom_pub
+    elif self.mode == SimulatorMode.IDEAL:
+        ball_odom_pub = rospy.Publisher('/ball/odom',
+                                        Odometry, queue_size=1)
+        return None, ball_odom_pub
+
+    return None, None
```

Comment:
I think this is also tame enough that we can leave it in the original init function. Function breakout should be reserved for cases where we want to hide the detail of modifying/mutating data. If we are just setting or declaring variables, then those operations should happen in the init function.